### PR TITLE
Fix sms-verification putting routes on other mod

### PR DESCRIPTION
**Pull request by henrahmagix**
>
The order of the script sourcing picked this up. Any other order and this
wouldn't have been seen, because it's not about one module trying to use a
dependency that's not registered.

With tests

@incuna/frontend @incuna/js Please merge, ta!

### CODE REVIEW

**Comment by henrahmagix** [comment]
Actually please hold off on merging until Travis is turned on. If anyone has the ability, please do so! https://travis-ci.org/profile/incuna

File: tests/unit/all/dependencies-spec.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+            it('should load the module without error', function () {
+                expect(function () {
+                    this.runModule();
+                }.bind(this)).not.toThrow();
+            });
+
+            it('should have routes', function () {
+                this.runModule();
+                expect(Object.keys(this.$route.routes)).not.toEqual([]);
```
Comment (maxpeterson) [inline tests/unit/all/dependencies-spec.js]: Shouldn't we test the actual routes of each module?

**Comment by maxpeterson** [comment]
travis is enabled

File: tests/unit/all/dependencies-spec.js (hunk 1 above)
Comment (henrahmagix) [inline tests/unit/all/dependencies-spec.js]: I've thought that before when I first started Angular testing, but it felt like too much like copy-pasting code, and that any route change would require a test change too. I guess as this is a library, the routes shouldn't be changed as often.

Should I test the route definitions, or just that the paths exist? I.e. 
```js
expect($route.routes['/some/route'].someProp).toBe(someValue);
```
or just 
```js
expect($route.routes['/some/route']).toBeDefined();
```

Comment (Minglee01) [inline tests/unit/all/dependencies-spec.js]: should you not test both?

as they could be named wrong but go where you want.
or named right but go to the wrong place.

File: Gruntfile.js [hunk 2]
```diff
@@ -84,17 +101,79 @@ module.exports = function (grunt) {
...
+                browsers: ['Chrome'],
+                logLevel: 'INFO',
+                autoWatch: true
+            },
+            ci: {
+                singleRun: true,
+                reporters: ['dots', 'coverage'],
+                browsers: ['Firefox']
+            }
+        }
```
Comment (wytrych) [inline Gruntfile.js]: Could you add the karma config in a separate file?

File: tests/unit/all/dependencies-spec.js (hunk 1 above)
Comment (henrahmagix) [inline tests/unit/all/dependencies-spec.js]: I removed this route testing in 6a5207d, and will implement better route testing later outside of this pull-request, now that this is testing the module isolation bug that was originally found.

File: Gruntfile.js (hunk 2 above)
Comment (henrahmagix) [inline Gruntfile.js]: Done in f7ee71f

**Comment by henrahmagix** [comment]
> travis is enabled

Thanks @maxpeterson!

@incuna/js This is now properly testing each module in isolation, and it fails if the injector throws an error when loading the module.

@wytrych would prefer we use the better module system of registering a module for every file (see [this gist](https://gist.github.com/henrahmagix/e0067ca707297ad952c6) for more information). I would like to do that, but considering the large change to dist files, I would prefer not to in this pull-request.

**Comment by henrahmagix** [comment]
This begins tests for #2

**Comment by henrahmagix** [comment]
@incuna/js Bump

**Comment by henrahmagix** [comment]
Thanks!

**Comment by henrahmagix** [comment]
Released as [`2.1.2`](https://github.com/incuna/angular-user-management/releases/tag/2.1.2)

**Pull request merged**